lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -512,6 +529,16 @@ def exec_module(self, **kwargs):
                         results.get('enable_accelerated_networking')))
                     changed = True
 
+                # We need to ensure that dns_servers are list like
+                dns_servers_res = results.get('dns_settings').get('dns_servers')
```

Comment:
Null check for dns_settings here?

Reply:
`dns_settings` should never be null as it comes from `network_client.network_interfaces.get()`
Certainly can test, but we could miss breaking API changes from Azure SDK if we do this...

Just for fun: The DNS settings does not conform to published API from a setting perspective BTW. I had an interesting conversation with the Azure Architect (or _an_ architect) about this.... turns out you cannot set anything except `dns_servers`, hence only this option. Well, you can set them, they are simply ignored.